fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/StringStatementRenderer.java

```diff
@@ -0,0 +1,303 @@
+    }
+
+    @Override
+    public String or(String left, String right) {
+        StringBuilder result = new StringBuilder();
+//        result.append(left);
+//        result.append(" OR ");
+//        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String equals(String left, String right) {
+        throw new IllegalStateException("not supported");
+//        StringBuilder result = new StringBuilder();
+//        result.append(left);
+//        result.append(" = ");
+//        result.append(right);
+//        return result.toString();
+    }
+
+    @Override
+    public String notEquals(String left, String right) {
+        StringBuilder result = new StringBuilder();
+//        result.append(left);
+//        result.append(" != ");
+//        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String not(String param) {
+        StringBuilder result = new StringBuilder();
+//        result.append("NOT ");
+//        result.append(param);
+        return result.toString();
+    }
+
+    @Override
+    public String exists(String statement) {
+        StringBuilder result = new StringBuilder();
+//        result.append("EXISTS (");
+//        result.append(statement);
+//        result.append(")");
+        return result.toString();
+    }
+
+    @Override
+    public String select(boolean distinct, SelectList selectList, FromClause fromClause, WhereClause whereClause, GroupByClause groupByClause, HavingClause havingClause,
+        OrderByClause orderByClause, PaginationClause paginationClause) {
+
+        StringExpNodeVisitor whereClauseRenderer = new StringExpNodeVisitor(this.translator, this.collectBindMarkersInto, this.pretty);
+
+        StringBuilder result = new StringBuilder();
+        if (this.pretty) {
+            result.append(NEWLINE).append("      "); // 6 spaces
+        }
+        result.append(SELECT);
+        if (distinct) {
+            result.append(" DISTINCT");
+        }
+        result.append(SPACE).append(selectList.toString());
+        result.append(SPACE);
+        if (this.pretty) {
+            result.append(NEWLINE).append("        "); // 8 spaces
+        }
+        result.append(FROM);
+        result.append(SPACE).append(fromClause.render(this));
+
+        if (whereClause != null && !whereClause.isEmpty()) {
+            if (this.pretty) {
+                result.append(NEWLINE).append("      ");
+            }
+            result.append(SPACE).append(WHERE);
+            result.append(SPACE).append(whereClause.visit(whereClauseRenderer));
+        }
+
+        if (groupByClause != null) {
+            if (this.pretty) {
+                result.append(NEWLINE).append("    ");
+            }
+            result.append(SPACE).append(groupByClause.toString());
+        }
+
+        if (havingClause != null) {
+            if (this.pretty) {
+                result.append(NEWLINE).append("    ");
```

Comment:
It'd be great if we could make this a constant, so know which spacing to use (since it's invisible)